src/KafkaFlow/Consumers/WorkersBalancers/ConsumerLagWorkerBalancer.cs

```diff
@@ -0,0 +1,158 @@
+namespace KafkaFlow.Consumers.WorkersBalancers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using KafkaFlow.Clusters;
+    using KafkaFlow.Configuration;
+
+    internal class ConsumerLagWorkerBalancer
+    {
+        private readonly IClusterManager clusterManager;
+        private readonly IConsumerAccessor consumerAccessor;
+        private readonly ILogHandler logHandler;
+        private readonly int totalConsumerWorkers;
+        private readonly int minInstanceWorkers;
+        private readonly int maxInstanceWorkers;
+
+        public ConsumerLagWorkerBalancer(
```

Comment:
Is it possible to provide some documentation here about the math used to find the number of workers based on the lags?

Reply:
Added summary comment in the class explaining the used math